website/src/blogs/2019-02-21-ha-vs-dr-and-ha-c2-b2-for-your-db.md

```diff
@@ -0,0 +1,96 @@
+2. **Build every layer** — and  every system,  so that it is built to fail.
+3. ***Build***** every layer** — do not think you have DR when you have HA; and don’t think you have HA just because you have one workload that spans several clusters.
+4. I**nfrastructure as code – **related to #3. Always; no black boxes. The desired state is in the repo. Yes, the control loop at the center of Kubernetes will make simple decisions to trade off workloads and behaviors to the best of its ability. Grok that, and insist on everything  by deploying this approach.
+5. **Simplicity**. Yes, this is the UNIX philosophy of do one thing and do it well. No, a human cannot grok the entire system in many cases , but they must be able to grok the behavior of each subsection.
+
+Let’s look at a few FAILURE examples. Below, I have included the entire Google slide deck that contains the discussed images for you to use and review. They are even animated in this slide deck!
+
+I’ll be working off this example of a Kubernetes environment. As you can see, we have two Kubernetes clusters, each of which have many pods. I show Pod A, B and C for each, but in reality, there would be many additional pods as well. We have some stateful workloads running across those clusters.
+![https://cdn-images-1.medium.com/max/800/0*vOLJvMtJlVTdBrNO](https://lh4.googleusercontent.com/n3WH9MopXX_jyDIfM2VoBJH2nhYeCl03zkd18bVwlMukZTMQsXdjOWviBMJp6ECs2O85zud3mkcy48NQJF3fkQiOolTY2l_l7_oPd67wLpn2VnGsW2QD1miHtZCPaByYM6mEbnyt)
+Now let’s assume you have similar environments on cloud A and data center B or similar. The idea is that we have multiple locations for similar multi-cluster Kubernetes.
+
+*It may be useful for these examples to keep in mind that the fundamental architecture of a Container Attached Storage solution like OpenEBS is to run within the clusters to provide storage to stateful workloads. Essentially, this is the piece that gives each workload its own file or block access (or s3 if you run Minio on top, for example) irrespective of the underlying cloud volumes, existing storage, disks or other hardware such as NVMe. *[*https://www.cncf.io/blog/2018/04/19/container-attached-storage-a-primer/*](https://www.cncf.io/blog/2018/04/19/container-attached-storage-a-primer/)
+![https://cdn-images-1.medium.com/max/800/0*Jyy9KksJm3vSUTFx](https://lh6.googleusercontent.com/GymcWw3kxRMK7FEsvDcW7K10r1R_qXRi4n7ebkE9NxpA1vItATFKD4XcLUmAWgcDbG4y-oG5kYWylk6a9O4f6yJgqdbfJZMJmumPcl0r3Gx4FEsKtcULB3dLS4KGCq1xXGU6MFv2)
+**Failure Condition #1: Cloud Volume or Disk Dies**
+
+Starting at the bottom of that stack, what should you do when a disk dies or a cloud volume becomes unresponsive? This is shown in by the red “X” in the image below.
+![https://cdn-images-1.medium.com/max/800/0*IqJ1INEeIgbopmfT](https://lh4.googleusercontent.com/RTnMj309bhhk4m6mjvYely45ESRGdePqJyKV-UVhA40CqvMknJHPSf9QG5tg4nSRL4b3v7LHX4fOtC-nE_2AatVe2MdSw7pPBLgpYjcdqJhoBJ3f7Z6iCQ0pczi6kTcl9lcENcOP)
```

Comment:
please fix this type of code `![https://cdn-images-1.medium.com/max/800/0*IqJ1INEeIgbopmfT](https://lh4.googleusercontent.com/RTnMj309bhhk4m6mjvYely45ESRGdePqJyKV-UVhA40CqvMknJHPSf9QG5tg4nSRL4b3v7LHX4fOtC-nE_2AatVe2MdSw7pPBLgpYjcdqJhoBJ3f7Z6iCQ0pczi6kTcl9lcENcOP)`, It's throughout blog

Reply:
@IsAmrish I haven't done any manipulation, it's coming for the old blog itself and the images also break there too.
check this blog for reference:
https://openebs.io/blog/running-openebs-on-custom-rancher-cluster/